server/src/main/java/org/opensearch/client/node/NodeClient.java

```diff
@@ -137,10 +143,11 @@ public String getLocalNodeId() {
     private <Request extends ActionRequest, Response extends ActionResponse> TransportAction<Request, Response> transportAction(
         ActionType<Response> action
     ) {
-        if (actions == null) {
+        if (actionRegistry == null) {
             throw new IllegalStateException("NodeClient has not been initialized");
         }
-        TransportAction<Request, Response> transportAction = actions.get(action);
+        // Get from action map if it exists
+        TransportAction<Request, Response> transportAction = actionRegistry.get(action);
```

Comment:
Since we are using feature flag to register actions dynamically. We should not this change directly on `main` branch. 
We can have it like:
```java
 TransportAction<Request, Response> transportAction = actions.get(action);
 if(feature_flag_enabled){
   TransportAction<Request, Response> dynamicTransportAction = actionRegistry.get(action);
 }
 ```

Reply:
Ah I missed the small `get` method in `ActionModule`. Looks fine. Ignore my above comment